warehouse/forklift/legacy.py

```diff
@@ -1092,6 +1115,27 @@ def file_upload(request):
                     f"{str(version)!r}.",
                 )
 
+            filename = os.path.basename(temporary_filename)
+
+            if meta.license_files and packaging.version.Version(
+                meta.metadata_version
+            ) >= packaging.version.Version("2.4"):
```

Comment:
Seems like the check for `metadata_version` is redundant here (and elsewhere), I don't think `meta.license_files` will exist if the version is < 2.4:

```
>>> Metadata.from_raw({"metadata_version": "2.3", "license_files": ["something"], "version": "6.6.6", "name": "throwaway"})
  + Exception Group Traceback (most recent call last):
  |   File "<stdin>", line 1, in <module>
  |   File "/home/di/.pyenv/versions/3.12.7/lib/python3.12/site-packages/packaging/metadata.py", line 752, in from_raw
  |     raise ExceptionGroup("invalid metadata", exceptions)
  | ExceptionGroup: invalid metadata (1 sub-exception)
  +-+---------------- 1 ----------------
    | packaging.metadata.InvalidMetadata: license-file introduced in metadata version 2.4, not 2.3
    +------------------------------------

>>> Metadata.from_raw({"metadata_version": "2.3", "version": "6.6.6", "name": "throwaway"}).license_files is None
True
```

Reply:
8c2859ae686a674eab04b543899984395be4abed